lib/msf/core/db_manager/import/qualys/asset.rb

```diff
@@ -21,15 +21,18 @@ def find_qualys_asset_ports(i,host,wspace,hobj,task_id)
 
   def find_qualys_asset_vuln_refs(doc)
     vuln_refs = {}
-    doc.elements.each("/ASSET_DATA_REPORT/GLOSSARY/VULN_DETAILS_LIST/VULN_DETAILS") do |vuln|
-      next unless vuln.elements['QID'] && vuln.elements['QID'].first
-      qid = vuln.elements['QID'].first.to_s
+    doc.xpath("/ASSET_DATA_REPORT/GLOSSARY/VULN_DETAILS_LIST/VULN_DETAILS").each do |vuln|
+      qid_el = vuln.xpath('QID')
+      next unless qid_el && qid_el.first
+      qid = vuln.xpath('QID').text
       vuln_refs[qid] ||= []
-      vuln.elements.each('CVE_ID_LIST/CVE_ID') do |ref|
-        vuln_refs[qid].push('CVE-' + /C..-([0-9\-]{9,})/.match(ref.elements['ID'].text.to_s)[1])
+      vuln.xpath('CVE_ID_LIST/CVE_ID').each do |ref|
+        id = ref.xpath("ID").first&.text
+        vuln_refs[qid].push("CVE-#{id}") if id
       end
-      vuln.elements.each('BUGTRAQ_ID_LIST/BUGTRAQ_ID') do |ref|
-        vuln_refs[qid].push('BID-' + ref.elements['ID'].text.to_s)
+      vuln.xpath('BUGTRAQ_ID_LIST/BUGTRAQ_ID').each do |ref|
+        id = ref.xpath("ID").first&.text
+        vuln_refs[qid].push("CVE-#{id}") if id
```

Comment:
Last I checked the BugTraq ID started with `BID-` not with `CVE-`, so this might need to be changed accordingly.